### DISCUSSION TITLE
Where can I find the documentation on fq query parameters like res_format, res_name, ecc.?

### DISCUSSION THREAD

**Discussion by aborruso** [ckan/ckan#7013] [Q&A] [answered]
Hi,
I thought I could not use to have the list of datasets that have at least one CSV resource.

Using ckanapi cli, I cannot run 

```
ckanapi search datasets -r https://dati.gov.it/opendata/ fq="format:CSV" rows=10
```

I must use `resource_search` action, to have results about resources and not datasets

```
ckanapi action resource_search query="format:CSV" -r https://dati.gov.it/opendata/ limit=1
```

But today I discovered that there is some sort of alias to look for resources inside the datasets: if I use the prefix `res_`, I can search using the resources field names, and search datasets by resources format:

```
ckanapi  -r https://dati.gov.it/opendata/ action package_search fq="res_format:CSV" rows=1
```

Am I right? And if yes, where can I find a documentation about that?

Thank you

**Comment by amercader** [comment, marked as answer]
You are right, you need to use `res_format` to filter by resource format. This is sadly not documented and should be part of a future search dedicated page (see https://github.com/ckan/ckan/issues/6915, and https://github.com/ckan/ckan/wiki/Solr-and-search)

For now, the next best thing is to look at the Solr schema to see which fields are indexed and available for searching: https://github.com/ckan/ckan/blob/f2eb20ebf9de21016fd3d15ed028be1cce8d447c/ckan/config/solr/schema.xml#L134

**Reply by aborruso** [reply]
> You are right, you need to use `res_format` to filter by resource format. This is sadly not documented and should be part of a future search dedicated page (see #6915, and https://github.com/ckan/ckan/wiki/Solr-and-search)

Thank you very much @amercader .

Opening [schema.xml](https://github.com/ckan/ckan/blob/f2eb20ebf9de21016fd3d15ed028be1cce8d447c/ckan/config/solr/schema.xml) it seems that I can search datasets filtering also by extras? Am I right?

If yes and I have these extras, how to search the datasets that have the key `contact_email` = `"opendata@sample.it"`.

Thank you

```json
"extras": [
        {
          "key": "contact_email",
          "value": "opendata@sample.it"
        },
        {
          "key": "contact_name",
          "value": "Lorem Ipsum"
        }
]
```

**Reply by amercader** [reply]
Extras are indexed both as `string` fields (as `contact_email`) and as `text` fields (as `extras_contact_email`). `string` fields are not processed by Solr so are good for exact matches. In your case you can just do:

```
  ckanapi  -r https://dati.gov.it/opendata/ action package_search fq="contact_email:opendata@sample.it" rows=1
```

**Reply by aborruso** [reply]
Thank you very much
